asserts/crypto.go

```diff
@@ -178,3 +230,55 @@ func parsePublicKey(pubKey []byte) (PublicKey, error) {
 	}
 	return WrapPublicKey(pubk), nil
 }
+
+// EncodePublicKey serializes a public key, typically for embedding in an assertion.
+func EncodePublicKey(pubKey PublicKey) ([]byte, error) {
+	return encodeOpenpgp(pubKey, "public key")
+}
+
+// PrivateKey is a cryptographic private/public key pair.
+type PrivateKey interface {
+	// PublicKey returns the public part of the pair.
+	PublicKey() PublicKey
+}
+
+type openpgpPrivateKey struct {
+	privk *packet.PrivateKey
+}
+
+func (opgPrivK openpgpPrivateKey) PublicKey() PublicKey {
+	return WrapPublicKey(&opgPrivK.privk.PublicKey)
+}
+
+func (opgPrivK openpgpPrivateKey) openpgpSerialize(w io.Writer) error {
+	return opgPrivK.privk.Serialize(w)
+}
+
+func parsePrivateKey(privKey []byte) (PrivateKey, error) {
+	pkt, err := parseOpenpgp(privKey, "private key")
+	if err != nil {
+		return nil, err
+	}
+	privk, ok := pkt.(*packet.PrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("expected private key, got instead: %T", pkt)
+	}
+	return openpgpPrivateKey{privk}, nil
+}
+
+// WrapPrivateKey returns a PrivateKey for database use out of a opengpg packet.PrivateKey.
+func WrapPrivateKey(privk *packet.PrivateKey) PrivateKey {
```

Comment:
s/Wrap/OpenPGP/?  Naming it "Wrap" makes it feel like PrivateKey is just a shell around the other, which might be technically right today, but it's not the goal of this abstraction. PrivateKey is our rock, and the openpgp variant is something we're choosing as the underlying implementation for the first version of it.